src/utils/mesh.js

```diff
@@ -242,10 +319,21 @@ async function checkAndInstallMeshPlugin (installedPlugins) {
  * @param installedPlugins
  */
 export async function createMesh (runAIOCommand, installedPlugins) {
-  const { saas, paas, catalog } = config.get('commerce.datasource')
-  const { org: githubOrg, repo: githubRepo } = config.get('commerce.github')
+  const shouldCreateMesh = await confirmAPIMeshCreation()
+
+  if (!shouldCreateMesh) {
+    aioLogger.debug('Not creating API Mesh - will use default environment')
+    return
+  }
+
+  await checkAndInstallMeshPlugin(installedPlugins)
```

Comment:
We don't need to do any of this unless `datasource.paas` or `datasource.saas` has values.

These having values indicate that the user selected either BYO or "select SaaS". If they select demo instance, we do not create a datasource.

Reply:
In that case, how are we going to skip mesh creation in case of a demo instance?

Reply:
the `if (paas || saas) { ... ` condition. 

If user selects demo, then `paas` and `saas` will not have any values. Thus this check will skip, so we don't create mesh.

We actually don't need to ask them whether to create the mesh or not - if they provide their own url (paas) or if they select saas, they should get a mesh. Period.

Reply:
Done